database/seeds/ModelSeeders/UserSeeder.php

```diff
@@ -0,0 +1,106 @@
+            $st1 = $u->statisticsOsu()->save(factory(App\Models\UserStatistics\Taiko::class)->create(['country_acronym' => $country_code, 'rank' => $rank2, 'rank_score_index' => $rank2]));
+            $st2 = $u->statisticsOsu()->save(factory(App\Models\UserStatistics\Fruits::class)->create(['country_acronym' => $country_code, 'rank' => $rank3, 'rank_score_index' => $rank3]));
+            $st3 = $u->statisticsOsu()->save(factory(App\Models\UserStatistics\Mania::class)->create(['country_acronym' => $country_code, 'rank' => $rank4, 'rank_score_index' => $rank4]));
+            // END USER STATS
+
+            // RANK HISTORY
+
+            // Create rank histories for all 4 modes
+            for ($c = 0; $c <= 3; $c++) {
+                switch ($c) {
+                    case 0: $rank = $st->rank; break;
+                    case 1: $rank = $st1->rank; break;
+                    case 2: $rank = $st2->rank; break;
+                    case 3: $rank = $st3->rank; break;
+                    default: $rank = $st->rank;
+                }
+
+                $hist = new App\Models\RankHistory;
+
+                $hist->mode = $c; // 0 = standard, 1 = taiko etc...
+
+                $play_freq = rand(10, 35); // How regulary the user plays (as a % chance per day)
+
+                // Start with current rank, and move down (back in time) to r0
+                $hist->r89 = $rank;
+
+                for ($i = 88; $i >= 0; $i--) {
+                    $r = 'r'.$i;
+                    $prev_r = 'r'.($i + 1);
+                    $prev_rank = $hist->$prev_r;
+                    // We wouldn't expect the user to improve every day
+                    $does_improve = $this->faker->boolean($play_freq);
+                    if ($does_improve === true) {
+                        $extreme_improvement = $this->faker->boolean(2); // chance of extreme improvement today
+                        if ($extreme_improvement === true) {
+                            $improvement_modifier = 1.5;
+                        } else {
+                            $improvement_modifier = $this->improvement_speeds[array_rand($this->improvement_speeds)];
+                        }
+                        $new_rank = round($hist->$prev_r * $improvement_modifier);
+                        if ($new_rank < 1) {
+                            $new_rank = 1;
+                        }
+                        // User rank will be modified by somewhere between 0.97 and 1.1 of current rank (realistic amount)
+                        $hist->$r = $new_rank;
+                    } else {
+                        $new_rank = round($hist->$prev_r * (rand(998, 999) / 1000));
+                        if ($new_rank < 1) {
+                            $new_rank = 1;
+                        }
+                        $hist->$r = $new_rank; // Slight decay of between 0.98 and 0.99
```

Comment:
A decay of 0.01-0.02% per day may be a little forgiving, it could perhaps be increased to 0.02-0.03%